emptyStrCmp/emptyStrCmp.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package emptystrcmp
+
+import (
+	"go/ast"
+	"go/token"
+	"golang.org/x/tools/go/analysis"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "emptyStrCmp",
+	Doc:  "check for len(s) == 0 and len(s) != 0 where s is string",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, file := range pass.Files {
+		ast.Inspect(file, func(node ast.Node) bool {
+			switch node.(type) {
```

Comment:
We can directly do `switch n := node.(type)` and remove the typecast in the next line.